### PR TITLE
Adds a fluff item for DesolateG 👌

### PR DESCRIPTION
**Pull request by FlattestGuitar**
It's a kit that turns a baton into a nicer looking scifi baton.

```
Ckey: DesolateG

Character name: Micheal Smith 

Jobs: Security Officer, Warden, Head of Security, Security Pod Pilot
```

*also fixes a teeny bit about another fluff item I made a while back*

### CODE REVIEW

**Comment by DesolateG** [comment]
yay!

can you post them the Gif @FlattestGuitar ?

**Comment by Aurorablade** [comment]
I have a concern that it may hide the fact you have a baton....

File: code/modules/customitems/item_defines.dm [hunk 1]
```diff
@@ -251,7 +295,7 @@
 		to_chat(user, "<span class='notice'>You modify the appearance of [target].</span>")
 
 		var/obj/item/clothing/head/welding/flamedecal/P = new(get_turf(target))
-		transfer_fingerprints_to(P)
+		target.transfer_fingerprints_to(P)
```
Comment (SamHPurp) [inline code/modules/customitems/item_defines.dm]: *gasp

**Comment by DesolateG** [comment]
@Aurorablade it's name, sound, and description match up with the stun baton.

Additionally it's been animated so that it's purpose of use is fairly clear to see.

[IMG]https://cdn.discordapp.com/attachments/239943874536144899/297023734299230208/Glowrod.gif[/IMG]

**Comment by DesolateG** [comment]
https://cdn.discordapp.com/attachments/239943874536144899/297023734299230208/Glowrod.gif

**Comment by DaveTheHeadcrab** [comment]
If noone else is gonna say it, I will. This baton definitely looks like a marital aid.

**Comment by Anticept** [comment]
I said the same thing when desolate first showed me.

**Comment by Xantholne** [comment]
@DaveTheHeadcrab I already did to the first version, although. This one looks a lot nicer now. Definately like the idea of a hi tech stunbaton though. You'd think after several hundred years we'd have something more advanced at least in some way.

**Comment by Fox-McCloud** [comment]
This is hugely overcomplicated for what it does, when you can just have it change the baton's icon state and make some changes to baton code to accommodate.

**Comment by DesolateG** [comment]
@Fox-McCloud that was the point.. but flattest coded it heh

**Comment by TravellingMerchant** [comment]
100% support from me!

No Bias here! Nope! _None._

**Comment by DesolateG** [comment]
@Necaladun @FreeStylaLT

**Comment by FlattestGuitar** [comment]
Changed the way it's all handled. 

I don't know why I felt like refactoring things with the fluff PRs was bad practice. Here it is now.

File: code/modules/customitems/item_defines.dm [hunk 2]
```diff
@@ -235,6 +235,29 @@
...
+	w_class = 2
+
+/obj/item/device/fluff/desolate_baton_kit/afterattack(atom/target, mob/user, proximity)
+	if(!proximity || !ishuman(user) || user.lying)
+		return
+
+	if(istype(target, /obj/item/weapon/melee/baton) && !istype(target, /obj/item/weapon/melee/baton/cattleprod))
+		to_chat(user, "<span class='notice'>You modify the appearance of [target].</span>")
+		//because batons use the initial() proc, we can't just swap the icon state, sadly
+		var/obj/item/weapon/melee/baton/the_baton = target
```
Comment (Fox-McCloud) [inline code/modules/customitems/item_defines.dm]: quick question; if you change the `icon` here too, will it still work; that way you can keep the baton icons in the custom items as opposed to polluting weapons.dmi.

File: code/modules/customitems/item_defines.dm [hunk 3]
```diff
@@ -235,6 +235,29 @@
...
 
+/obj/item/device/fluff/desolate_baton_kit //DesolateG: Michael steampunk_witch
+	name = "stun baton converstion kit"
+	desc = "Some sci-fi looking parts for a stun baton."
+	icon = 'icons/obj/custom_items.dmi'
+	icon_state = "scifikit"
+	w_class = 2
+
+/obj/item/device/fluff/desolate_baton_kit/afterattack(atom/target, mob/user, proximity)
+	if(!proximity || !ishuman(user) || user.lying)
```
Comment (Fox-McCloud) [inline code/modules/customitems/item_defines.dm]: `user.incapacitated()` instead of `user.lying`

**Comment by FlattestGuitar** [comment]
@Fox-McCloud addressed your thingies.

**Comment by Fox-McCloud** [comment]
Last request; move this item's in-hands, also, to their own file (I don't expect you to do that for every single other fluff item, in this PR).

I'm tired of fluff items polluting general sprite sheets; if makes it more difficult to remove, and if a downstream wants to kill all fluff items dead, then it's far easier for them, as well.

**Comment by FlattestGuitar** [comment]
@Fox-McCloud, addressed. Again.

I'll bother tossing all the old icons into the new file eventually, for now just the baton is in there.

**Comment by DesolateG** [comment]
whew.. my fluff triggered refractoring.

**Comment by Fox-McCloud** [comment]
Huzzah; much cleaner.

**Comment by Fox-McCloud** [comment]
@Necaladun @FreeStylaLT

**Comment by Birdtalon** [comment]
It looks like a sword.

That is all.

**Comment by Regen1** [comment]
No, it doesn't look like a baton, which is a weapon that should be **clearly** distinguishable on first sight.

**Comment by Elysian-Prince** [comment]
You've got my upvote. Looks nice; a bit like a sword with rings at the top, but nice regardless. You'ev got my vote.

**Comment by DesolateG** [comment]
@Elysian-Prince based off of 
[the Demolition Man glowrod](https://www.youtube.com/watch?v=oOadr-Ts4J8)

**Comment by taukausanake** [comment]
Wait so that's what that movie was called? I loved that movie! It was so bad and enjoyable.

Yeah it does look like a sword. Maybe the rings are a bit too long?

**Comment by FreeStylaLT** [comment]
It looks a bit too much like a sword and less like a baton, even when comparing it to the source material.
I would suggest shortening the pommel, guard and trimming the.. stick part.
Something along the lines of this:
img
Note: Not an actual sprite artist, just made some adjustments after seeing the source material + original baton sprite

**Comment by DesolateG** [comment]
That can be arranged. @TravellingMerchant

**Comment by FlattestGuitar** [comment]
Updated the sprite.

It's now more 🍆, less ⚔️

**Comment by Fox-McCloud** [comment]
@FreeStylaLT ?

**Comment by FreeStylaLT** [comment]
👌

**Comment by DesolateG** [comment]
Thanks guys

**Pull request merged**